### PR TITLE
[2.9.x] Move to newer CodeStyle packages and fix build warnings

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <!-- Use the correct compiler version -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
     <!-- Dependencies from https://github.com/dotnet/roslyn -->
-    <MicrosoftNETCoreCompilersPackageVersion>3.8.0</MicrosoftNETCoreCompilersPackageVersion>
+    <MicrosoftNETCoreCompilersPackageVersion>3.9.0-2.final</MicrosoftNETCoreCompilersPackageVersion>
```
Comment (mavasani) [inline eng/Versions.props:25]: @Evangelink I would suggest you make this change even in https://github.com/dotnet/roslyn-analyzers/pull/4555 to ensure none of the IDE warning violations are missed from being fixed.

Comment (sharwell) [reply eng/Versions.props:25]: ⚠️ This will make the minimum version of VS for working on this repository 16.9 Preview 2

Comment (mavasani) [reply eng/Versions.props:25]: Hmm, I am fine to mark this PR as blocked till 16.9 public release.

I understand this requirement on Roslyn repo due to large contributor base. I think that is an acceptable requirement on this repo though, but don't see any real rush in doing this.

**Review by Evangelink — APPROVED** [review]
LGTM

File: eng/Versions.props (hunk 1 above)
Comment (Evangelink) [reply eng/Versions.props:25]: Thanks for the warning @sharwell

**Comment by mavasani** [comment]
The code style fixes in this PR are addressed in https://github.com/dotnet/roslyn-analyzers/pull/4595. Move to 3.9 CodeStyle layer needs to wait till VS2019 16.9 release, which is few months out. I'll close this PR for now.

**Pull request closed without merging**